pkg/controller/jobframework/integrationmanager.go

```diff
@@ -231,12 +241,14 @@ func EnableIntegration(name string) {
 // Mark the frameworks identified by names and return a revert function.
 func EnableIntegrationsForTest(tb testing.TB, names ...string) func() {
 	tb.Helper()
-	old := manager.enabledIntegrations.Clone()
+	old := manager.getEnabledIntegrations()
 	for _, name := range names {
 		manager.enableIntegration(name)
 	}
```

Comment:
It would be more accurate to obtain the lock once for this entire operation. But because this is just for tests, it's fine.